src/test/ui/return/return-impl-trait.rs

```diff
@@ -0,0 +1,24 @@
+trait Trait {}
+impl Trait for () {}
+
+// this works
+fn foo() -> impl Trait {
+    ()
+}
+
+fn bar<T: Trait + std::marker::Sync>() -> T
```

Comment:
Please also add the NOTEs to the test so it's easier to see what it's testing.

Reply:
The compiler outputs a lot of notes, some even have line numbers in them. Is there a way to not have to write all of them but just the most important ones?

Reply:
Not that I know of, unfortunately.